writer.go

```diff
@@ -136,10 +148,7 @@ func (w *Writer) Debug(m string) (err error) {
 func (w *Writer) writeAndRetry(p Priority, s string) (int, error) {
 	pr := (w.priority & facilityMask) | (p & severityMask)
 
-	w.Lock()
-	defer w.Unlock()
-
-	if w.conn != nil {
+	if w.getConn() != nil {
```

Comment:
True. I'll make it so we hold the lock long enough but avoid holding it while we're writing.